Source/WebCore/rendering/RenderBlockFlow.cpp

```diff
@@ -1367,6 +1375,50 @@ bool RenderBlockFlow::shouldTrimChildMargin(MarginTrimType marginTrimType, const
     return lastInFlowChildBox() == &child;
 }
 
+void RenderBlockFlow::trimFloatBlockEndMargins(LayoutUnit blockFormattingContextInFlowContentHeight)
+{
+    ASSERT(m_floatingObjects && createsBlockFormattingContext());
+    auto computeFloatMarginAfterStartLocationInBlockFormattingContext = [&](const RenderBox& floatBox) {
+        // Start with the location of the float within its containing block. If this containing
+        // block is the BFC in which we are trimming, then we can use this as the starting location
+        // since this is the float's positin within the BFC. If not, add the containing block's location
+        // to the float's location. This new value should be the location of the float relative to this
+        // new containing block. Keep doing this until we find the containing block that establishes
+        // a BFC
+        auto startLocation = floatBox.y();
+        auto containingBlock = floatBox.containingBlock();
+        while (!containingBlock->createsBlockFormattingContext()) {
+            startLocation += containingBlock->y();
+            containingBlock = containingBlock->containingBlock();
+        }
+        return startLocation + floatBox.height();
```

Comment:
How does this work with transforms and other non-trivial translates like position: relative? is this writing-mode aware?

Reply:
(though maybe the fact that we stay within BFC makes it a lot simpler than something like in localToContainerQuad/mapLocalToContainer, I'd still check out those functions as you do a very similar translate here.

Reply:
Changed to use `logicalTopForChild` and `logicalHeightForChild` so that we get correct value according to the containing block's writing mode